Improve/expand contribution instructions

waldyrious opened the pull request:
Prompted by the discussion in https://github.com/waldyrious/computefreely/pull/6#discussion_r1379112594, and based on my own exploration of this repository in the course of [various cleanup changes I've been making](https://github.com/waldyrious/computefreely/commits/main?after=cb6a0afd098afc70bf64ec90c0021308d75313f3+0), here are some updates to the contribution documentation to clarify some ambiguous bits, add extra information, fix a some typos, and overall make this project slightly easier to contribute to. 

This is an important step towards making the project ready for moving to an organization to be community-maintained, as it cements some practices and guidelines that were not yet fully fleshed out.

I'll leave this PR open for a few days in case there's any feedback. I'd be happy to hear others' opinions on whether the updated text is clear and useful enough. /cc @cassidyjames who started the discussion linked above, as well as @rugk and @CaKrome, who reacted in #2.

waldyrious committed:
Improve/expand contribution instructions

rugk commented on CONTRIBUTING.md:
```diff
@@ -35,28 +39,31 @@ notes:
+#### Distribution metadata
 
-The `notes` section contains pertinent info about a distribution that show's up in the 'Notes' section on each post.
+The `notes` section contains pertinent info about the distribution itself that shows up in the "Notes" section on each post.
```
```suggestion
The `notes` section contains pertinent info about the distribution itself that shows up in the “Notes” section on each post.
```

waldyrious replied:
Thanks for the review! I generally prefer using straight quotes in markdown source, since those are easier to produce in most keyboard layouts, and renderers can often auto-convert to smart quotes. Can you explain your reasoning for preferring the latter?

rugk replied:
If the renderer does it, then fine yeah. If they don't, then this is needed. AFAIK Jekyll does not do it out-of-the box so, this is more practical and less theory.

The quotes I use are just the correct typographical quotes. If you are into that stuff (or ever wrote anything with LaTeX :wink:) you see these typographical errors everywhere. :upside_down_face:

waldyrious replied:
Heheh gotcha. However, this file is not rendered on the site, so Jekyll's rendering is not relevant, but rather what GitHub uses to render markdown here, as we browse the source code. [This](https://github.com/waldyrious/computefreely/blob/main/CONTRIBUTING.md) is what it looks like.

Do you feel the lack of typographical quotes in that page is too conspicuous? I would prefer to not include them in the source code as that makes it harder to edit (I, for once, have to copy existing instances because I don't have a convenient way to input them — and indeed [most people don't](https://en.wikipedia.org/wiki/Quotation_mark#Keyboard_layouts)), and the straight quotes in the rendered text don't bother me.

That said, if you do feel strongly about this, I'd be open to adding them anyway as my objection is relatively light.

rugk replied:
Yeah GitHub does not special rendering. I know it's unpractical, you have to pay attention etc. but it's just proper typesetting. Most people don't care, if you read newspaper articles you will see they (obviously should) care etc.

But as we are in a Linux project some tips/FYI:
* for such special characters [the compound key is actually very useful](https://askubuntu.com/questions/70784/how-can-i-enable-compose-key)
* for these quotes, I at least in a German QUERTZ layout do have (by default) a third special characters at the <kbd>v</kbd> („), <kbd>b</kbd> (“), <kbd>n</kbd> (”) keys:
   ![image](https://github.com/waldyrious/computefreely/assets/11966684/fb6a91a9-8930-4259-92dd-ab8f34e68d12)
   I am personally quite accustomed to using these for such inputs.

waldyrious replied:
Huh. I just checked and indeed in my Portuguese QWERTY layout, `“` and `”` are also in the <kbd>V</kbd> and <kbd>B</kbd> keys:

![image](https://github.com/waldyrious/computefreely/assets/478237/36eab4a5-d877-4b67-95a8-fc946b12a37a)

So I suppose it's less of a chore to keep using these than I assumed. I'll go ahead and integrate your suggestions :)

rugk commented on CONTRIBUTING.md:
```diff
@@ -35,28 +39,31 @@ notes:
 - `based-on` is a list of distributions that this one is based on (if any)
-- `desktops` is a list of all default desktops arranged in an array, i.e. if a distribution provides multiple spins that are of different desktop environments you list those here
+- `desktops` is an array of all default desktop environments provided by the distribution
 - `install` is the mode of installation — either "graphical", "command line" or "live"
```
```suggestion
- `install` is the mode of installation — either “graphical”, “command line” or “live”
```

rugk reviewed (approved):
Generally LGTM.

waldyrious committed:
Apply suggestions from code review

Use typographically correct, curved quote symbols instead of the straight ones.

Co-authored-by: rugk <rugk+git@posteo.de>

waldyrious commented:
I've added the suggested changes, @rugk — thanks for the review and feedback, and sorry it took me so long to respond each time!

Since nobody else offered feedback in over a month, I'll go ahead and merge :)
